src/Sylius/Bundle/ProductBundle/Form/EventSubscriber/ProductOptionFieldSubscriber.php

```diff
@@ -45,7 +59,7 @@ public function preSetData(FormEvent $event)
         $form = $event->getForm();
 
         /** Options should be disabled for configurable product if it has at least one defined variant */
-        $disableOptions = (null !== $product->getFirstVariant()) && (false === $product->hasVariants());
+        $disableOptions = (null !== $this->defaultVariantResolver->getVariant($product)) && (false === $product->hasVariants());
```

Comment:
Actually current logic can be simplified to `$disableOptions = $product->hasVariants()`